doc/man3/X509_check_purpose.pod

```diff
@@ -0,0 +1,54 @@
+=pod
+
+=head1 NAME
+
+X509_check_purpose - Checks certificate I<x's> purpose
+
+=head1 SYNOPSIS
+
+#include <openssl/x509v3.h>
+
+int X509_check_purpose(X509 *x, int id, int ca)
+
+=head1 DESCRIPTION
+
+This function checks if B<certificate> I<x> was created with the B<purpose>
+represented by I<id>. If I<ca> is non-zero, then certificate I<x> is
+checked to determine if it's a possible CA with various levels of certainty
+possibly returned.
+
+Below are the potential ID's that can be checked:
+# define X509_PURPOSE_SSL_CLIENT        1
+# define X509_PURPOSE_SSL_SERVER        2
+# define X509_PURPOSE_NS_SSL_SERVER     3
+# define X509_PURPOSE_SMIME_SIGN        4
+# define X509_PURPOSE_SMIME_ENCRYPT     5
+# define X509_PURPOSE_CRL_SIGN          6
+# define X509_PURPOSE_ANY               7
+# define X509_PURPOSE_OCSP_HELPER       8
+# define X509_PURPOSE_TIMESTAMP_SIGN    9
+
+=head1 RETURN VALUES
+
+For non-CA checks
+1 if the B<certificate> was created to perform the B<purpose> represented by I<id>
+0 if the B<certificate> was not created to perform the B<purpose> represented by I<id>
```

Comment:
If you want to do it in style, I would suggest this:

```
For non-CA checks

=over 4

=item 1 if the B<certificate> was created to perform the B<purpose> represented by I<id>

=item 0  if the B<certificate> was not created to perform the B<purpose> represented by I<id>

=back
```

Reply:
Fixed